From 4fbe8e868065b39618a3e7eec351815773155201 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:10:33 -0400
Subject: [usb] Revert: pl2303: add two comments concerning the supported baud rates with HX chips

Message-id: <1410887463-20674-562-git-send-email-dzickus@redhat.com>
Patchwork-id: 95509
O-Subject: [RHEL7 PATCH 561/591] Revert "usb: pl2303: add two comments concerning the supported baud rates with HX chips"
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 233c3dda5cd1bb26fa871b94db17627117e51026
Author: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date:   Fri Nov 1 09:18:25 2013 -0700

    Revert "usb: pl2303: add two comments concerning the supported baud rates with HX chips"

    This reverts commit c23bda365dfbf56aa4d6d4a97f83136c36050e01.

    Revert all of the pl2303 changes that went into 3.12-rc1 and -rc2 as
    they cause regressions on some versions of the chip.  This will all be
    revisited for later kernel versions when we can figure out how to handle
    this in a way that does not break working devices.

    Reported-by: Mika Westerberg <mika.westerberg@linux.intel.com>
    Cc: Frank Schafer <fschaefer.oss@googlemail.com>
    Acked-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/pl2303.c b/drivers/usb/serial/pl2303.c
index b4cfb86..fcc1326 100644
--- a/drivers/usb/serial/pl2303.c
+++ b/drivers/usb/serial/pl2303.c
@@ -285,12 +285,6 @@ static int pl2303_baudrate_encode_direct(int baud, enum pl2303_type type,
      4800, 7200, 9600, 14400, 19200, 28800, 38400,
      57600, 115200, 230400, 460800, 614400, 921600,
      1228800, 2457600, 3000000, 6000000 };
- /*
-  * NOTE: The PL2303HX (tested with rev. 3A) also supports the following
-  * baud rates: 128000, 134400, 161280, 201600, 268800, 403200, 806400.
-  * As long as we are not using this encoding method for them, there is
-  * no point in complicating the code to support them.
-  */
  int i;
 
  /* Set baudrate to nearest supported value */
@@ -404,12 +398,6 @@ static void pl2303_encode_baudrate(struct tty_struct *tty,
   *    => supported by all chip types
   * 2) Divisor based method: encodes a divisor to a base value (12MHz*32)
   *    => supported by HX chips (and likely not by type_0/1 chips)
-  *
-  * NOTE: Although the divisor based baud rate encoding method is much
-  * more flexible, some of the standard baud rate values can not be
-  * realized exactly. But the difference is very small (max. 0.2%) and
-  * the device likely uses the same baud rate generator for both methods
-  * so that there is likley no difference.
   */
  if (type != HX)
   baud = pl2303_baudrate_encode_direct(baud, type, buf);
-- 
1.7.1